### ISSUE TITLE
Type Error in hot-toast-ref.d.ts

### ISSUE THREAD

**Issue by Van-Dame** [ngneat/hot-toast#16]
## I'm submitting a...

<pre><code>[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

Adding library to angular app and running build/serve throws following error:

<pre><code>Error: node_modules/@ngneat/hot-toast/lib/hot-toast-ref.d.ts:27:5 - error TS2416: Property 'close' in type 'HotToastRef' is not assignable to the same property in base type 'HotToastRefProps'.
  Type '(closeData?: { dismissedByAction: boolean; } | undefined) => void' is not assignable to type '(closeData?: { dismissedByAction?: boolean | undefined; } | undefined) => void'.
    Types of parameters 'closeData' and 'closeData' are incompatible.
      Type '{ dismissedByAction?: boolean | undefined; } | undefined' is not assignable to type '{ dismissedByAction: boolean; } | undefined'.
        Type '{ dismissedByAction?: boolean | undefined; }' is not assignable to type '{ dismissedByAction: boolean; }'.
          Types of property 'dismissedByAction' are incompatible.
            Type 'boolean | undefined' is not assignable to type 'boolean'.
              Type 'undefined' is not assignable to type 'boolean'.

27     close(closeData?: {
       ~~~~~
</code></pre>

## Expected behavior

<!-- Describe what the desired behavior would be. -->
App should build normally.

## Minimal reproduction of the problem with instructions

1. `npm i -S @ngneat/overview @ngneat/hot-toast`
2. Add `HotToastModule.forRoot()` to `app.module.ts`
3. run build/serve

## Environment

<pre><code>Angular CLI: 11.1.2
Node: 15.6.0
OS: win32 x64

**Comment by shhdharmen** [comment]
Have you created project in strict mode?

**Comment by Van-Dame** [comment]
I believe Angular CLI 11 creates app in strict mode by default.

**Comment by nartc** [comment]
Drafted a PR that enables strict mode. I need some extra eyes to take a look at the PR to determine the changes aren't containing breaking change.

**Comment by shhdharmen** [comment]
> I believe Angular CLI 11 creates app in strict mode by default.

@Van-Dame No, CLI does not create project in strict mode by default. It's false [by default](https://angular.io/cli/new#options).

Anyways, I also created 2 workspaces, one with strict mode and one without it, both built successfully.

Can you please create a repo with the code in which build is failing?

**Comment by Van-Dame** [comment]
@shhdharmen I was under (false) impression that with Angular 11 they did generate app in strict mode by default, so thank you for remediating that.

For creating an MRE, I created a separate new project in strict mode and this time indeed, it worked as expected. However I noticed that the version of hot-toast library was updated to 1.0.3. I uninstalled and installed 1.0.1 and  the bug was back. Therefore, the bug was in that specific version and the latest update takes care of it.
